web-ng/src/app/services/router/router.service.ts

```diff
@@ -75,4 +84,50 @@ export class RouterService {
   getObservationId$(): Observable<string | null> {
     return this.observationId$!;
   }
+
+  /**
+   * Returns the current URL fragment, parsed as if their were normal HTTP
+   * query parameter key/value pairs.
+   */
+  private getFragmentParams(): HttpParams {
+    const fragment = this.activatedRoute!.snapshot.fragment;
```

Comment:
! -> ?  for undefined type

Reply:
`!` is correct - we want to throw an error if `getFragmentParams()` was called before `init()` (i.e. activatedRoute was not yet set).